### PR TITLE
Convert wiki to markdown docs

### PR DESCRIPTION
**Pull request by uglide**
Minimalistic conversion of asciidocs from wiki into markdown.

### CODE REVIEW

**Review by tishun — APPROVED** [review]
Looks awesome!

File: docs/Advanced-usage.md [hunk 1]
```diff
@@ -0,0 +1,2686 @@
...
+Client instances with using shared `ClientResources` (i.e.
+`ClientResources` passed using the constructor/`create`-method) won’t
+shut down the `ClientResources` on their own. The `ClientResources`
+instance needs to be shut down once it’s not used anymore.
+
+### Configuration settings
+
+The basic configuration options are listed in the table below:
+
+| Name                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | Method                       | Default                |
```
[the remark this answers is not in the document]
Comment (uglide) [reply docs/Advanced-usage.md]: I fully agree, still I have no easy/quick fix for that. Let's create an issue to track it.

**Comment by uglide** [comment]
Doc site URL: https://redis.github.io/lettuce/

**Comment by uglide** [comment]
CC: @andy-stark-redis

File: docs/Advanced-usage.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tishun) [reply docs/Advanced-usage.md]: Totally

[the remark this answers is not in the document]
Comment (tishun) [reply docs/Advanced-usage.md]: #2954

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: Publish Docs
+on:
+  push:
+    branches: ["main", "markdown_docs"]
```
Comment (tishun) [inline .github/workflows/docs.yml:4]: TODO : We might want to have different versions per branch / release

**Review by tishun — APPROVED** [review]
pom.xml changes look okay

**Comment by tishun** [comment]
Could you also add the missing words to the spellcheck dictionary?

**Pull request merged**